### PR TITLE
Monica/ui fixes

### PR DESCRIPTION
**Pull request by myo3**

### CODE REVIEW

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 1]
```diff
@@ -286,7 +283,12 @@ class RouteOptionsViewController: UIViewController, UITableViewDelegate, UITable
     // MARK: Process data
 
     func searchForRoutes() {
-        if let startingDestination = searchFrom as? CoordinateAcceptor, let endingDestination = searchTo as? CoordinateAcceptor{
+        // If no date is set then date should be same as today's date
+        if routeSelection.datepickerButton.titleLabel?.text == "Leave now" {
```
Comment (mattbarker016) [inline TCAT/Controllers/RouteOptionsViewController.swift]: Is there a more stable way to check this?

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 2]
```diff
@@ -419,6 +421,18 @@ class RouteOptionsViewController: UIViewController, UITableViewDelegate, UITable
     func showDatepicker(sender: UIButton){
         view.bringSubview(toFront: datePickerOverlay)
         view.bringSubview(toFront: datePickerView)
+        
+        // set up date on datepicker view
+        if routeSelection.datepickerButton.titleLabel?.text == "Leave now" {
+            let now = Date()
+            datePickerView.setDatepickerDate(date: now)
```
Comment (mattbarker016) [inline TCAT/Controllers/RouteOptionsViewController.swift]: don't need to set the variable, can just put in parameters

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 3]
```diff
@@ -517,26 +531,35 @@ class RouteOptionsViewController: UIViewController, UITableViewDelegate, UITable
         NotificationCenter.default.removeObserver(self, name: .reachabilityChanged, object: reachability)
     }
     
+    override var prefersStatusBarHidden: Bool {
```
Comment (mattbarker016) [inline TCAT/Controllers/RouteOptionsViewController.swift:533]: just return isBannerShown

File: TCAT/Model/Route.swift [hunk 4]
```diff
@@ -198,6 +198,10 @@ class Route: NSObject, JSONDecodable {
         }
     }
     
+    func isWalkingRoute() -> Bool {
+        return routeSummary.count == 2 && routeSummary.first?.nextDirection == .walk
```
Comment (mattbarker016) [inline TCAT/Model/Route.swift]: I know backend was sometimes returning odd walking directions with 3 steps, maybe to be safe, just run a .all functional check to make sure all directions are walk maybe

File: TCAT/Views/DatepickerView.swift [hunk 5]
```diff
@@ -31,20 +32,29 @@ class DatepickerView: UIView {
...
         styleCancelButton()
         styleDoneButton()
-        styleDisclaimerLabel()
+//        styleDisclaimerLabel()
 
         setDatepickerSettings()
-        setSegmentedControl(withItems: segmentedControlOptions)
+        
+        setSegmentedControl(timeTypeSegmentedControl, withItems: timeTypeSegmentedControlOptions)
+        let leaveAt = 0
```
Comment (mattbarker016) [inline TCAT/Views/DatepickerView.swift:52]: if you don't use this again, might just want to set the next line to be 0 and add a comment

File: TCAT/Controllers/RouteOptionsViewController.swift (hunk 1 above)
Comment (myo3) [reply TCAT/Controllers/RouteOptionsViewController.swift]: unfoturnately not

File: TCAT/Views/DatepickerView.swift [hunk 6]
```diff
@@ -56,14 +66,37 @@ class DatepickerView: UIView {
     
     // MARK: Segement Control
     
-    func segmentedControlValueChanged(segmentControl: UISegmentedControl) {
-        let leaveNow = 0
-        if segmentedControl.selectedSegmentIndex == leaveNow {
-            datepicker.date = Date()
-            segmentedControl.selectedSegmentIndex = prevSelectedIndex
+    func leaveNowSegmentedControlValueChanged(segmentControl: UISegmentedControl) {
```
Comment (mattbarker016) [inline TCAT/Views/DatepickerView.swift:69]: (Possibly) apply similar comments mentioned above (`let now = Date()` and `let leaveNow = 0`)

**Review by mattbarker016 — COMMENTED** [review]
Good stuff otherwise!

**Pull request merged**